src/components/hoc/withSession.tsx

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { redirect } from 'next/navigation'
+import { useSupabase } from '../SupabaseProvider'
+import { Session } from '@supabase/supabase-js'
+
+export type WithSessionProps = {
+  session: Session
+}
+
+export function withSession<TProps = WithSessionProps>(
```

Comment:
I remember in our call the hoc didn't work, but I tried it now locally and it worked just fine on first try, like this at `messages/new/page`:

```
import { withSession } from '@/components/hoc/withSession'

// ....


// export default NewMessagePage
export default withSession(NewMessagePage)
```

I even added a console.log inside `withSession` to double-check.
Was there anything you changed in the meantime?

Reply:
I didn't change anything 😅 tried again no luck, I must be doing something wrong, we can take a look on thursday!